### PR TITLE
[Do Not Merge] ElasticSearch Flexibility

### PR DESCRIPTION
**Pull request by platinummonkey**
**DO NOT MERGE**
This PR is meant to flush out all the bits necessary for a flexible uniform ES chart that we can use from 1.x->6.x+

**What this PR does / why we need it:**
We wanted some more flexibility on the runtime options for ElasticSearch. Mostly refactors elasticsearch.yml into it's own template as a number of additional settings were added.

This will support 1.x line if custom images are provided.

**Which issue this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged):**

fixes #

**Special notes for your reviewer:**

This is still a WIP, trying to get some quick feedback.

### CODE REVIEW

File: incubator/elasticsearch/README.md [hunk 1]
```diff
@@ -71,11 +71,16 @@ The following table lists the configurable parameters of the elasticsearch chart
 | `cluster.xpackEnable`                | Writes the X-Pack configuration options to the configuration file   | `false`                              |
 | `cluster.config`                     | Additional cluster config appended                                  | `{}`                                 |
 | `cluster.env`                        | Cluster environment variables                                       | `{}`                                 |
+| `cluster.routing_attributes`         | Cluster routing attributes                                          | `nil`                                |
```
Comment (pietdaniel) [inline incubator/elasticsearch/README.md]: this should be camel cased to be consistent

File: incubator/elasticsearch/README.md [hunk 2]
```diff
@@ -71,11 +71,16 @@ The following table lists the configurable parameters of the elasticsearch chart
...
+| `cluster.gclogEnable`                | Enable GC Logging                                                   | `false`                              |
 | `client.name`                        | Client component name                                               | `client`                             |
 | `client.replicas`                    | Client node replicas (deployment)                                   | `2`                                  |
 | `client.resources`                   | Client node resources requests & limits                             | `{} - cpu limit must be an integer`  |
 | `client.priorityClassName`           | Client priorityClass                                                | `nil`                                |
 | `client.heapSize`                    | Client node heap size                                               | `512m`                               |
+| `client.java.opts_equal`             | Client node java `-<key>=<value>` options                           | `{}`                                 |
```
Comment (pietdaniel) [inline incubator/elasticsearch/README.md]: this should be camel cased to be consistent

File: incubator/elasticsearch/README.md [hunk 3]
```diff
@@ -101,6 +108,8 @@ The following table lists the configurable parameters of the elasticsearch chart
 | `data.resources`                     | Data node resources requests & limits                               | `{} - cpu limit must be an integer`  |
 | `data.priorityClassName`             | Data priorityClass                                                  | `nil`                                |
 | `data.heapSize`                      | Data node heap size                                                 | `1536m`                              |
+| `data.java.opts_equal`               | Data node java `-<key>=<value>` options                             | `{}`                                 |
```
Comment (pietdaniel) [inline incubator/elasticsearch/README.md]: this should be camel cased to be consistent

File: incubator/elasticsearch/README.md [hunk 4]
```diff
@@ -90,6 +95,8 @@ The following table lists the configurable parameters of the elasticsearch chart
 | `master.nodeSelector`                | Node labels for master pod assignment                               | `{}`                                 |
 | `master.tolerations`                 | Master tolerations                                                  | `{}`                                 |
 | `master.heapSize`                    | Master node heap size                                               | `512m`                               |
+| `master.java.opts_equal`             | Master node java `-<key>=<value>` options                           | `{}`                                 |
```
Comment (pietdaniel) [inline incubator/elasticsearch/README.md]: this should be camel cased to be consistent

**Review by pietdaniel — CHANGES_REQUESTED** [review]
This PR does a lot, can we split into smaller functional components.

**Comment by platinummonkey** [comment]
Updated title and message to reflect this PR is not meant to be merged, but to flush out ideas. Later smaller PR's will be created to make this more digestible.

**Comment by platinummonkey** [comment]
@pietdaniel to take the lead! closing this

**Pull request closed without merging**